staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -259,7 +264,16 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationv1api.APIServ
 		servicePort:      *apiService.Spec.Service.Port,
 		serviceAvailable: apiregistrationv1apihelper.IsAPIServiceConditionTrue(apiService, apiregistrationv1api.Available),
 	}
-	if r.proxyTransport != nil && r.proxyTransport.DialContext != nil {
+	if r.egressSelector != nil {
```

Comment:
same comments I made here: https://github.com/kubernetes/kubernetes/pull/85870#discussion_r369279460

* pass this around via options struct, not by adding parameters to functions
* use a no-op implementation so that this nil check isn't needed.

cc @Jefftree

Reply:
@lavalamp hmm it's already passed via the `proxyHandler` struct. Or do you mean adding a `APIOptionsStruct` as a parameter to the function? (Which is adding a parameter 😅, cause adding the field to `apiService` seems weird. 

For the second point would that be a helper function taking the egress selector + the dialer and returning either the egress selector dialer or the passed one? (`nil` check would still be made in the helper though right?)

Reply:
@lavalamp: Per your second point, there are a lot of other egress endpoints implemented without the no-op implementation that have already been merged. I can make the change requested in the other PR (#85870), but can we look at this PR independent of the no-op and do another refactor PR later?

cc @caesarxuchao

Reply:
agree that no-op could be changed across all egress in its own PR

Reply:
RE. the second point, @jefftreee will send a follow-up refactor PR after this one and #85870 are merged.